distributed/tests/test_scheduler.py

```diff
@@ -2389,6 +2389,33 @@ async def test_idle_timeout(c, s, a, b):
     pc.stop()
 
 
+@gen_cluster(client=True)
+async def test_idle_during_update_graph(c, s, a, b):
+    class UpdateGraphTrackerPlugin(SchedulerPlugin):
+        def start(self, scheduler):
+            self.scheduler = scheduler
+            self.idle_during_update_graph = None
```

Comment:
[Non-blocking nit] This is totally fine as is but maybe don't even assign `self.idle_during_update_graph` until inside the `update_graph` plugin hook? `assert plugin.idle_during_update_graph is None` _could_ mean that the `update_graph` hook didn't run at all. 

```suggestion
```

Reply:
I've slightly adjusted this such that `idle_during_update_graph` is initialized to `None` and set to `True` or `False` during the hook.